pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -274,9 +276,12 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 				// Remote side closed connection.
 				return receiveError
 			}
-			err = s.initConnectionNode(discReq, con)
-			if err != nil {
-				return err
+			if discReq.Node != nil {
```

Comment:
So if we get 3 requests, one with node, then without, then with node, do we want to set `node` again the 3rd time? Or just the first time?

Reply:
Doesn't really matter since they should be the same.

Reply:
Why not only parse once since they are identical?

Reply:
I don't really follow this code so I want to make it correct first, optimize later. Envoy will stop sending nodes eventually with every request.

Reply:
I think the behavior is correct - as long as envoy doesn't send a node with an empty or incomplete value in the next requests. I would protect against this by checking if any of the node elements has been populated, or adding a bool to make sure it is parsed only once.

Reply:
Added an extra guard for Id != ""